src/app/lib/utilities/preprocessor/rules/cpsAssetPage/addBylineBlock/index.test.js

```diff
@@ -0,0 +1,151 @@
+import deepClone from 'ramda/src/clone';
+import addBylineBlock from '.';
+
+const styInput = {
+  metadata: { type: 'STY' },
+  promo: {
+    byline: {
+      name: 'Foo Bar',
+      title: 'BBC News Title',
+      persons: [
+        {
+          name: 'Foo Bar',
+        },
+      ],
+    },
+  },
+  content: {
+    model: {
+      blocks: [
+        {
+          type: 'headline',
+        },
+        {
+          type: 'image',
+        },
+        {
+          type: 'timestamp',
+        },
+      ],
+    },
+  },
+};
+
+describe('addBylineBlock', () => {
+  it('should add a block of type byline after the first headline block', async () => {
+    const input = deepClone(styInput);
+    const expected = {
+      metadata: { type: 'STY' },
+      promo: {
+        byline: {
+          name: 'Foo Bar',
+          title: 'BBC News Title',
+          persons: [
+            {
+              name: 'Foo Bar',
+            },
+          ],
+        },
+      },
+      content: {
+        model: {
+          blocks: [
+            {
+              type: 'headline',
+            },
+            {
+              type: 'byline',
+              model: {
+                blocks: [
+                  {
+                    name: 'Foo Bar',
+                    title: 'BBC News Title',
+                  },
+                ],
+              },
+            },
+            {
+              type: 'image',
+            },
+            {
+              type: 'timestamp',
+            },
+          ],
+        },
+      },
+    };
+
+    expect(addBylineBlock(input)).toEqual(expected);
+  });
+
+  it('should add a block of type byline using the persons array when byline has no title and name properties', async () => {
+    const input = deepClone(styInput);
+    delete input.promo.byline.name;
+    delete input.promo.byline.title;
+
+    input.promo.byline.persons[0].function = 'BBC News Title';
+    const expected = {
+      metadata: { type: 'STY' },
+      promo: {
+        byline: {
+          persons: [
+            {
+              name: 'Foo Bar',
+            },
+          ],
+        },
+      },
+      content: {
+        model: {
+          blocks: [
+            {
+              type: 'headline',
+            },
+            {
+              type: 'byline',
+              model: {
+                blocks: [
+                  {
+                    name: 'Foo Bar',
+                    title: 'BBC News Title',
+                  },
+                ],
+              },
+            },
+            {
+              type: 'image',
+            },
+            {
+              type: 'timestamp',
+            },
+          ],
+        },
+      },
+    };
+
+    expected.promo.byline.persons[0].function = 'BBC News Title';
```

Comment:
Any reason this isn't included after https://github.com/bbc/simorgh/pull/5408/files#diff-43a47b8493bb9c80b7416a64bdfee057R93

Reply:
@andrewcfc `function` is a reserved keyword and eslint blocks `"function"`

Reply:
ohh that makes sense, maybe add a comment, no worries though!